conda_package_streaming/transmute.py

```diff
@@ -89,18 +77,41 @@ def transmute(
                 else:
                     tar_get.addfile(member)
 
+            info_tar.close()
             pkg_tar.close()
-            pkg_stream.close()
 
-            info_tar.close()
-            info_stream.close()
+            info_size = info_file.tell()
+            pkg_size = pkg_file.tell()
+
+            info_file.seek(0)
+            pkg_file.seek(0)
 
-        with conda_file.open(f"info-{file_id}.tar.zst", "w") as info_file:
-            info_file.write(info_io.getvalue())
+        with zipfile.ZipFile(
+            os.path.join(path, f"{file_id}.conda"),
+            "x",  # x to not append to existing
+            compresslevel=zipfile.ZIP_STORED,
+        ) as conda_file:
+
+            # Use a maximum of one Zstd compressor, stream_writer at a time to save memory.
+            data_compress = compressor()
+
+            with conda_file.open(
+                f"pkg-{file_id}.tar.zst", "w"
+            ) as pkg_file_zip, data_compress.stream_writer(
+                pkg_file_zip, size=pkg_size, closefd=False
+            ) as pkg_stream:
+                shutil.copyfileobj(pkg_file._file, pkg_stream)
+
+            with conda_file.open(
+                f"info-{file_id}.tar.zst", "w"
+            ) as info_file_zip, data_compress.stream_writer(
+                info_file_zip, size=info_size, closefd=False
+            ) as info_stream:
+                shutil.copyfileobj(info_file._file, info_stream)
```

Comment:
It would be worthwhile forcing this to be > the SpooledTemporaryFile rollover size.

Is the "assert identical conda/.tar.bz2 code only in the other package https://github.com/conda/conda-package-handling/blob/main/src/conda_package_handling/validate.py

Reply:
Luckily we always check for identical packages in the `cph transmute` command. https://github.com/conda/conda-package-handling/blob/main/src/conda_package_handling/api.py#L187

Reply:
Can force the rollover with the rollover() method, would that help you for this:
> It would be worthwhile forcing this to be > the SpooledTemporaryFile rollover size.

Reply:
I mean we need to test this with some > 10MB packages so that we can prove it works with, or without rollover.

Reply:
(I tried it with the 20MB libllvm14-14.0.6-h7ec7a93_1.conda by transmuting to and from .tar.bz2 and they were identical, as shown by `cph transmute`'s built-in checks, and by extracting them to separate directories and running `diff -r`.)

Reply:
Oh, and it didn't have a metadata.json, whoops.

Reply:
Whoops :)